client/src/components/Instructions/index.js

```diff
@@ -92,6 +92,100 @@ export default class Instructions extends Component {
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            1. Ensure Docker is installed on your computer.
+          </div>
+          <div className={styles.code}>
+            <code>
+              <a href="https://docs.docker.com/install/">https://docs.docker.com/install/</a>
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            2. In another terminal, run the gas relay docker container with ganache, wait a few seconds for it to sping up (run inside your zepkit folder).
+          </div>
+          <div className={styles.code}>
+            <code>
+              npm explore tabookey-gasless npm run gsn-dock-relay-ganache
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            3. In a different terminal, init your project with ZeppelinOS (run inside your zepkit folder).
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos init zepkit
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            4. Add the GaslessCounter contract to your project (Note that a compilation warning will be thrown, you can ignore this).
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos add GaslessCounter
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            5. Connect with your local blockchain by opening a session (Note that we automatically prefilled your ganache account).
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos session --network development --from {addressDefault} --expires 3600
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            6. Let's compile the Gasless Counter contract.
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos push --deploy-dependencies
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            7. We create an instance of our contract and deploy it.
+          </div>
+          <div className={styles.code}>
+            <code>
+              zos create GaslessCounter --init initialize
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            8. Hook up your Counter to your Relay.
+          </div>
+          <div className={styles.code}>
+            <code>
+              npx truffle@4.1.13 migrate
```

Comment:
I tried running this with truffle 5, and it throws an error because the behaviour of `Contract.at` changed between versions 4 and 5: it now returns a promise to the contract instance, instead of the contract directly. Rewriting the migration like this seems to work fine in truffle 5:
```
    let counterAddr = Counter.networks[Object.keys(Counter.networks)[0]].address;
    let hubAddr = networks[network].relayHubAddr;
    RelayHub.at(hubAddr).then(hub => hub.depositFor(Counter.address, { value:1e18 }));
    console.log("Initializing Counter's Hub");
    Counter.at(counterAddr).then(counter => counter.init_hub(hubAddr))
```

Reply:
Still, I'd suggest **not** relying on truffle migrations for this: the migrations are general to the whole project, while this is something very specific of the gasless counter tutorial. Maybe having a separate js script, or just instructing the user to run this in their console...?

Reply:
I will work on extending the tutorial into a truffle console set of commands so the user is more aware of what setup needs to be done in the background.